pkg/pillar/cas/containerd.go

```diff
@@ -785,14 +795,18 @@ func (c *containerdCAS) IngestBlobsAndCreateImage(reference string, root types.B
 	rootBlobSha := fmt.Sprintf("%s:%s", digest.SHA256, strings.ToLower(root.Sha256))
 	mediaType := root.MediaType
 	imageHash, err := c.GetImageHash(reference)
-	if err != nil || imageHash == "" {
-		logrus.Infof("IngestBlobsAndCreateImage: creating reference: %s for rootBlob %s", reference, rootBlobSha)
+	if errors.Is(err, ErrImageNotFound) || imageHash == "" {
```

Comment:
Is this a different error than the one exported by containerd?

Reply:
No, it is just there to abstract away from the concrete error from containerd so that other parts of pillar don't get tightly coupled on containerd.

An alternative could be to not return an error from `GetImageHash` and only use an empty `imageHash` as indicator that the image has not been found.